### PR TITLE
Issue15869 coverity

### PR DESCRIPTION
**Pull request by zycz**

### CODE REVIEW

File: tests/subsys/event_manager_proxy/src/data.c [hunk 1]
```diff
@@ -55,25 +56,27 @@ static void test_data_response(void)
...
 {
 	struct data_big_event *event = new_data_big_event();
+	struct data_big_content rand_test_data;
 
 	for (size_t n = 0; n < DATA_BIG_EVENT_BLOCK_SIZE; ++n) {
 		event->block[n] = sys_rand32_get();
 	}
 
 	memset(&data_big_response, 0, sizeof(data_big_response));
+	memcpy(&rand_test_data.block, &event->block, sizeof(data_big_response));
```
Comment (rakons) [inline tests/subsys/event_manager_proxy/src/data.c:79]: Put sizeof of the actual target. This one is risky if anything changes.

**Review by rakons — CHANGES_REQUESTED** [review]

File: tests/subsys/event_manager_proxy/src/data.c [hunk 2]
```diff
@@ -38,13 +38,14 @@ static struct data_big_content data_big_response;
 static void test_data_response(void)
```
Comment (MarekPieta) [inline tests/subsys/event_manager_proxy/src/data.c:38]: Fix commit message:
`Test copies data ...`

File: tests/subsys/event_manager_proxy/src/data.c [hunk 3]
```diff
@@ -82,7 +85,7 @@ static void test_data_big_response(void)
 	int err = k_sem_take(&waiting_response_sem, K_SECONDS(1));
 	zassert_ok(err, "No data response event received");
 
-	zassert_mem_equal(&data_big_response.block, &event->block, sizeof(event->block),
+	zassert_mem_equal(&data_big_response.block, &rand_test_data.block, sizeof(event->block),
```
Comment (MarekPieta) [inline tests/subsys/event_manager_proxy/src/data.c:88]: `sizeof(data_big_response.block)`? Event is no longer used here.

File: tests/subsys/event_manager_proxy/src/data.c (hunk 1 above)
Comment (MarekPieta) [reply tests/subsys/event_manager_proxy/src/data.c:79]: Should we also somehow ensure that both pointed data blocks have the same length (keep in mind these are separate types)?

**Review by MarekPieta — APPROVED** [review]

**Review by MarekPieta — APPROVED** [review]

**Pull request merged**